databases/week1/homework.sql

```diff
@@ -0,0 +1,69 @@
+-- Active: 1719743290311@@0.0.0.0@3306@USER
+
+--1. Find out how many tasks are in the task table
+SELECT COUNT(*) FROM task; -- Answer:35
+
+--2. Find out how many tasks in the task table do not have a valid due date
+SELECT COUNT(*) FROM task WHERE due_date IS NULL; -- Answer:8
+
+--3. Find all the tasks that are marked as done
+SELECT *
+FROM task
+JOIN status ON task.status_id = status.id
+JOIN `user` ON task.user_id = `user`.id
+WHERE status.name = 'Done';
+
+--4. Find all the tasks that are not marked as done
+
+SELECT *
+FROM task
+JOIN status ON task.status_id = status_id
+JOIN `user` ON task.user_id = user_id
+WHERE status.name != 'Done';
+
+--5. Get all the tasks, sorted with the most recently created first
+SELECT *
+FROM task
+ORDER BY created DESC;
+
+--6. Get the single most recently created task
+SELECT *
+FROM task
+ORDER BY created DESC
+LIMIT 1;
+
+--7. Get the title and due date of all tasks where the title or description contains database
+SELECT *
+FROM task
+WHERE title LIKE '%database%' OR description LIKE '%database%';
+
+--8. Get the title and status (as text) of all tasks
+
+SELECT user.name, task.title, status.name
+FROM task
+JOIN status ON task.status_id = status_id
+JOIN `user` ON task.user_id = user_id;
+
+--9. Get the name of each status, along with a count of how many tasks have that status
+SELECT status.name, COUNT(*)
+FROM task
+JOIN status ON task.status_id = status_id
+GROUP BY status.name;
+
+--10. Get the names of all statuses, sorted by the status with most tasks first
+SELECT status.name, COUNT(*) AS task_count
+FROM status
+LEFT JOIN task ON status_id = task.status_id
+GROUP BY status.name
+ORDER BY task_count DESC;
```

Comment:
Same as above. Also, a Left Join is not necessary, a JOIN (default for inner join) is enough.